### PR TITLE
[CodeGen] TwoAddressInstructionPass: Control NumVisited limit via command line option

### PR DESCRIPTION
**Pull request by AZero13**
Pulled out of comment made on https://github.com/llvm/llvm-project/pull/80627 - to simplify further investigation into visit limits.

### CODE REVIEW

**Comment by AtariDreams** [comment]
@RKSimon Is this good?

File: llvm/lib/CodeGen/TwoAddressInstructionPass.cpp [hunk 1]
```diff
@@ -80,6 +80,13 @@ EnableRescheduling("twoaddr-reschedule",
                    cl::desc("Coalesce copies by rescheduling (default=true)"),
                    cl::init(true), cl::Hidden);
 
+// Limit the number of rescheduling visits to dependent instructions.
+// FIXME: Arbitrary limit to reduce compile time cost.
+static cl::opt<unsigned>
+    MaxVisits("twoaddr-visit-limit", cl::Hidden, cl::init(10),
```
Comment (topperc) [inline llvm/lib/CodeGen/TwoAddressInstructionPass.cpp:86]: Should we include "reschedule" in the option name? "twoaddr-reschedule-visit-limit"?

Comment (AtariDreams) [reply llvm/lib/CodeGen/TwoAddressInstructionPass.cpp:86]: I feel like I am not the person who should make that decision

Comment (RKSimon) [reply llvm/lib/CodeGen/TwoAddressInstructionPass.cpp]: `twoaddr-reschedule-visit-limit` sgtm

**Review by RKSimon — CHANGES_REQUESTED** [review]
Please add test coverage to an existing test file where you add an additional RUN with a different twoaddr-reschedule-visit-limit value

**Pull request closed without merging**